examples/docker/solara/app.py

```diff
@@ -1,31 +1,111 @@
-
-    # Display messages based on the current word count and word limit.
-    if word_count >= int(word_limit.value):
-        solara.Error(
-            f"With {word_count} words, you passed the word limit of {word_limit.value}."
-        )
-    elif word_count >= int(0.8 * word_limit.value):
-        solara.Warning(
-            f"With {word_count} words, you are close to the word limit of {word_limit.value}."
-        )
+    text = State.text.value
+    package = State.package.value
+    results = State.results.value
+    stopwords = State.stopwords.value
+
+    with solara.Sidebar():
+        with solara.Card("Controls", margin=0, elevation=0):
+            with solara.Column():
+                with solara.Row():
+                    solara.Button("Sample dataset", color="primary", text=True, outlined=True, on_click=State.load_sample)
+                    solara.Button("Clear dataset", color="primary", text=True, outlined=True, on_click=State.reset)
+                FileDrop(on_file=State.load_from_file, on_total_progress=lambda *args: None, label="Drag a .txt file here")
+                if text is not None:
+                    solara.Select(label="Package", value=State.package, values=['PyTextRank', 'Yake'])
+                    solara.InputInt("Number of results", value=State.results, continuous_update=True)
+                    solara.SliderInt(label="Minimum Ngram", value=State.min_ngrams, min=1, max=10)
+                    solara.SliderInt(label="Maximum Ngram", value=State.max_ngrams, min=1, max=10)
+                    solara.Checkbox(label="Remove stopwords", value=State.stopwords)
+                solara.Markdown("Hosted in [Ploomber Cloud](https://ploomber.io/)")
+
+    if text is not None:
+        nlp = spacy.load("en_core_web_sm")
+        if stopwords:
+            doc = nlp(text)
+            text = ' '.join([token.text for token in doc if not token.is_stop])
+
+        if package == "PyTextRank":
+            # add PyTextRank to the spaCy pipeline
+            nlp.add_pipe("textrank")
+            doc = nlp(text)
+            phrases_data = []
+            for phrase in doc._.phrases:
+                phrase_ngrams = len(phrase.text.split())
+                if State.min_ngrams.value <= phrase_ngrams <= State.max_ngrams.value:
+                    phrases_data.append([phrase.text, phrase.rank])
+            if phrases_data:
+                phrases_data = phrases_data[:results]
+                # Create a DataFrame
+                keywords_df = pd.DataFrame(phrases_data, columns=['Text', 'Rank']).sort_values(by='Rank', ascending=False)
+                solara.DataFrame(keywords_df)
+                solara.FileDownload(keywords_df.to_csv(index=False), label=f"Download keywords",
+                                    filename="keyowrds.csv")
+            else:
+                solara.Info("No keywords found")
+        elif package == "Yake":
```

Comment:
when I click on Yake, I see "no keywords found"

Reply:
![image](https://github.com/ploomber/doc/assets/989250/17ef4360-1c85-4a64-9e1b-3c3839517b68)

switching to the other package shows the keywords as well

Reply:
This is happening because on removal of stopwords Yake was not able to extract any keywords:
<img width="923" alt="Screenshot 2023-10-17 at 8 05 37 PM" src="https://github.com/ploomber/doc/assets/29096135/c4d5e1a2-6e47-4596-9afc-0088e2f177b2">